ansible_base/rbac/api/serializers.py

```diff
@@ -257,9 +266,19 @@ def _get_related(self, obj) -> dict[str, str]:
             return {}
         related_fields = {}
         actor_cls = self.Meta.model
+
+        # Use ansible_id if available, otherwise fall back to pk
+        actor_identifier = obj.pk
+        try:
+            if hasattr(obj, 'resource') and obj.resource:
+                actor_identifier = str(obj.resource.ansible_id)
+        except ObjectDoesNotExist:
+            # Resource doesn't exist, stick with pk
+            logger.error(f"Resource does not exist for {self.Meta.model} {obj.pk}")
```

Comment:
Should this actually be an error or just a warning since we are going to continue on?
```suggestion
            logger.warn(f"Resource does not exist for {self.Meta.model} {obj.pk}, unable to get ansible_id, will attempt to use PK instead")
```

Reply:
Yes, I agree it can be a warn. We are only here when `hasattr(obj, 'resource')` is true. In that case the model _is_ registered with the resource registry, but the signals failed to do their expected job of creating an associated resource object.

The related link will still kind of work, but on the higher-level of UI-API interaction, the user experience will break.

I'll reword the message, but will diverge somewhat from your suggestion.